src/main/java/org/thoughtcrime/securesms/notifications/NotificationCenter.java

```diff
@@ -370,6 +372,27 @@ public void notifyReaction(int accountId, int contactId, int msgId, String react
       });
     }
 
+    public void notifyWebxdc(int accountId, int contactId, int msgId, String text) {
+      Util.runOnAnyBackgroundThread(() -> {
+        NotificationPrivacyPreference privacy = Prefs.getNotificationPrivacy(context);
+        if (!privacy.isDisplayContact() || !privacy.isDisplayMessage()) {
+          return; // showing "New Message" is wrong, just do nothing.
+        }
+
+        DcContext dcContext = context.dcAccounts.getAccount(accountId);
+        DcMsg dcMsg = dcContext.getMsg(msgId);
+        JSONObject info;
+        if(dcMsg.getType() == DcMsg.DC_MSG_WEBXDC) {
+          info = dcMsg.getWebxdcInfo();
+        } else { // info message, get from parent xdc
+          info = dcMsg.getParent().getWebxdcInfo();
```

Comment:
getParent() can be null, so maybe be more resilient here, a try/catch around things or so